src/lib/suit.ts

```diff
@@ -1,12 +1,16 @@
 const NAMESPACE = 'ais';
 
-export const component = (componentName: string) => ({
-  descendantName,
-  modifierName,
-}: {
+interface SuitNames {
   descendantName?: string;
   modifierName?: string;
-} = {}) => {
+}
+
+type SuitSelector = (names?: SuitNames) => string;
+
+export const component = (componentName: string): SuitSelector => ({
+  descendantName,
+  modifierName,
+}: SuitNames = {}) => {
```

Comment:
Again, do we want to allow inline types? It might be handy.

Reply:
Interfaces and types are useful when you repeat, but would feel over the top for me if necessary everywhere

Reply:
There is no rules that restrict the usage of inline types, so you're free to use what is best for your usage. In this specific case I just found clearer to use an interface and a type.